From d16ed50b48403d40b8125d9303d0cae7cff664dc Mon Sep 17 00:00:00 2001
From: Michael Adam <obnox@samba.org>
Date: Tue, 15 Mar 2016 12:36:59 +0100
Subject: [PATCH 1060/1096] smbd:smb2: add request_counters_updated to the
 smbd_smb2_request struct

This will be used to keep track of whether the outstanding request
counters have been updated in the dispatch, so that the reply
code can act accordingly.

Signed-off-by: Michael Adam <obnox@samba.org>
Reviewed-by: Jeremy Allison <jra@samba.org>
(cherry picked from commit ae6967ea3e39a1a5401be4a4c969b467dd22dce4)

BUG: https://bugzilla.samba.org/show_bug.cgi?id=11809
(cherry picked from commit a962decf6a8b2be58221bd43f20fdebb7b167cbd)
---
 source3/smbd/globals.h | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/source3/smbd/globals.h b/source3/smbd/globals.h
index 15f29a8..9e3e95c 100644
--- a/source3/smbd/globals.h
+++ b/source3/smbd/globals.h
@@ -727,6 +727,13 @@ struct smbd_smb2_request {
 	struct files_struct *compat_chain_fsp;
 
 	/*
+	 * Keep track of whether the outstanding request counters
+	 * had been updated in dispatch, so that they need to be
+	 * adapted again in reply.
+	 */
+	bool request_counters_updated;
+
+	/*
 	 * The sub request for async backend calls.
 	 * This is used for SMB2 Cancel.
 	 */
-- 
2.9.0

